File: 1750-Minimum-Length-of-String-After-Deleting-Similar-Ends/go/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println("Hello World")
	s := "aabccabba"
	fmt.Println(minimumLength(s))

}

func minimumLength(s string) int {

	left := 0
	right := len(s) - 1
	// fmt.Println(s[0])

	for left < right {
		if s[left] == s[right] {
			same := s[left]
			for left <= right && s[left] == same {
				left += 1
			}
			for left <= right && s[right] == same {
				right -= 1
			}
		} else {
			break
		}
	}
	return right - left + 1
}

File: 1750-Minimum-Length-of-String-After-Deleting-Similar-Ends/py/main.py
def minimumLength(self, s: str) -> int:
    left = 0
    right = len(s) - 1

    while left < right :
        if s[left] == s[right] :
            same = s[left]
            while (left<=right and s[left] == same) :
                left += 1
            while left<=right and s[right] == same :
                right -= 1
        else :
            break
            
    return (right-left +1)  

def main() {
    print(minimumLength("aabccabba"))
    print(minimumLength("cabaabac"))
}

main()
